launcher/modManager/cmodlist.cpp

```diff
@@ -208,11 +212,25 @@ CModEntry CModList::getMod(QString modname) const
 	}
 	else
 	{
-		if(conf.canConvert<QVariantMap>())
+		if(!conf.toMap().isEmpty())
 			settings = conf.toMap();
 		else
 			settings.insert("active", conf);
 	}
+	
+	if(settings["active"].toBool())
+	{
+		QString rootPath = path.section('/', 0, 1);
+		if(path != rootPath)
+		{
+			conf = getValue(modSettings, rootPath);
+			auto confMap = conf.toMap();
+			if(!conf.isNull() && !confMap["active"].toBool())
```

Comment:
First you convert node to map, then you check if the node type is not null - the logic looks incorrect

Reply:
confMap will be empty if cons is null

Reply:
after double-check I agree that this is ok.

But I've just noticed another issue on this line: `operator[]` [inserts value if it doesn't exist](https://doc.qt.io/qt-5/qmap.html#operator-5b-5d). I guess such behavior is not intended/needed and you should use `QMap::value()` method instead. Alternatively, declare `confMap` as `const`, then you can safely use `operator[]` without side effects.

Reply:
ok, I'm a have a god-feeling that those changes will break this very fragile fix :)